src/tooltips.cpp

```diff
@@ -60,16 +60,56 @@ void tooltip::init_label()
 	rect game_canvas = video::game_canvas();
 	unsigned int border = 10;
 
+	bool mydebug=false;
+
+	rect huge;
+	huge.h=1000000;
+	huge.w=1000000;
+
 	label.set_font_size(font_size);
 	label.set_color(font::NORMAL_COLOR);
-	label.set_clip_rect(game_canvas);
-	label.set_width(text_width);
+	label.set_clip_rect(huge);  
+	label.set_width(text_width);   // If tooltip will be too tall for game_canvas, this could be scaled up appropriately
 	label.set_alignment(font::LEFT_ALIGN);
 	label.set_bg_color(bgcolor);
 	label.set_border_size(border);
 
 	label.create_texture();
 
+	point lsize = label.get_draw_size();
+	int new_text_width = text_width;
+	if(lsize.y > game_canvas.h*0.95) {
+		// The multiplier (currently 2) is hard to select, since making a texture wider is no guarantee that there will be fewer lines of text.
+		// Could just use something really big, or incrementally increase it until the texture fits vertically (or width maxes out)?
+		//
+		// This block of text is just
+		// as tall as the other one.
+		//
+		// This block of text is just as tall as the other
+		// one.
+		new_text_width = 2 * text_width * static_cast<float>(static_cast<float>(lsize.y)/game_canvas.h);
```

Comment:
You probably don't need two float casts here.

Reply:
true